ruby/lib/microservice_logging.rb

```diff
@@ -14,22 +14,16 @@ def initialize(service_name:,
     @clock = clock
     @output = output
     @scoped_properties = scoped_properties
-    @event_loggers = {}
     @events = events
-    @events.each do |method_name|
-      event_type = method_name.to_s
-      @event_loggers[method_name] =
-        JsonEventLogger.new(:service_name => @service_name,
-                            :clock => @clock,
-                            :output => @output,
-                            :scoped_properties => @scoped_properties,
-                            :event_type => event_type)
-    end
   end
 
-  def method_missing(event)
-    super unless @events.include? event
-    @event_loggers[event]
+  def method_missing(event_or_method_name)
+    super unless respond_to_missing?(event_or_method_name)
+    event_loggers.fetch(event_or_method_name)
+  end
+
+  def respond_to_missing?(event_or_method_name)
```

Comment:
Why does adding an extra method for a what is already a 2-line method make things more readable?